lib/ansible/modules/source_control/gitlab_project_variable.py

```diff
@@ -0,0 +1,246 @@
+requirements:
+  - python >= 2.7
+  - python-gitlab python module
+extends_documentation_fragment:
+  - auth_basic
+options:
+  state:
+    description:
+      - create or delete project variable.
+      - Possible values are present and absent.
+    default: present
+    type: str
+    choices: ["present", "absent"]
+  api_token:
+    description:
+      - Gitlab access token with api permissions.
+    required: true
+    type: str
+  project:
+    description:
+      - The path and name of the project
+    required: true
+    type: str
+  purge:
+    description:
+      - When set to true, all variables which are not presented in the task will be deleted.
+    default: false
+    required: false
+    type: bool
+  vars:
+    description:
+      - A list of key value pairs
+    default: {}
+    required: false
+    type: dict
+'''
+
+
+EXAMPLES = '''
+- name: Set or update some CI/CD variables
+  gitlab_project_variable:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: markuman/dotfiles
+    purge: False
+    vars:
+      ACCESS_KEY_ID: abc123
+      SECRET_ACCESS_KEY: 321cba
+
+- name: delete one variable
+  gitlab_project_variable:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: markuman/dotfiles
+    state: absent
+    vars:
+      ACCESS_KEY_ID: abc123
+'''
+
+RETURN = '''
+msg:
+  description: Success or failure message
+  returned: always
+  type: str
+  sample: "Success"
+
+result:
+  description: json parsed response from the server
+  returned: always
+  type: dict
+
+error:
+  description: the error message returned by the Gitlab API
+  returned: failed
+  type: str
+  sample: "Failed to connect to Gitlab server: 401: 401 Unauthorized"
+
+project_variable:
+  description: three lists of the variablenames which were added, updated or removed.
+  returned: always
+  type: dict
+'''
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils._text import to_native
+from ansible.module_utils.api import basic_auth_argument_spec
+
+
+GITLAB_IMP_ERR = None
+try:
+    import gitlab
+    HAS_GITLAB_PACKAGE = True
+except Exception:
+    GITLAB_IMP_ERR = traceback.format_exc()
+    HAS_GITLAB_PACKAGE = False
+
+
+class gitlab_project_variables(object):
+
+    def __init__(self, module):
+        self.repo = gitlab.Gitlab(module.params['api_url'],
+                                  private_token=module.params['api_token'])
+        self.project = self.get_project(module.params['project'])
+        self._module = module
+
+    def auth(self):
+        self.repo.auth()
+
+    def get_project(self, project_name):
+        return self.repo.projects.get(project_name)
+
+    def list_all_project_variables(self):
+        raw_variable_list = self.project.variables.list()
+        retval = []
+        if len(raw_variable_list) > 0:
+            for item in raw_variable_list:
+                retval.append(item.get_id())
```

Comment:
I've found a bug :D I guess I introduced it while adding the check_mode for the update method().  
I added one more integration test for it.  
However, this will reduce the api requests of the gitlab inuse, but it needs some reworks of my "main" loop.